xclim/indices/_helpers.py

```diff
@@ -0,0 +1,266 @@
+    if method == "spencer":
+        return (
+            1.0001100
+            + 0.034221 * np.cos(da)
+            + 0.001280 * np.sin(da)
+            + 0.000719 * np.cos(2 * da)
+            + 0.000077 * np.sin(2 * da)
+        )
+
+
+def cosine_of_solar_zenith_angle(
+    declination: xr.DataArray, lat: xr.DataArray, stat: str = "integral"
+) -> xr.DataArray:
+    """Cosine of the solar zenith angle.
+
+    The solar zenith angle is the angle between a vertical line (perpendicular to the ground)
+    and the sun rays. This function computes a daily statistic of its cosine : its integral
+    from sunrise to sunset or the average over the same period. Based on [Kalogirou14]_.
+
+    Parameters
+    ----------
+    declination : xr.DataArray
+        Daily solar declination. See :py:func:`solar_declination`.
+    latitude : xr.DataArray
+        Latitude.
+    stat : {'integral', 'average'}
+        Which daily statistic to return. If "integral", this returns the integral of the
+        cosine of the zenith angle from sunrise to sunset. If "average", the integral is
+        divided by the "duration" from sunrise to sunset.
+
+
+    Returns
+    -------
+    Cosine of the solar zenith angle, [rad] or [dimensionless]
+        If stat is "integral", dimensions can be said to be "time" as the integral is on
+        the hour angle. For seconds, multiply by the number of seconds in a comple day
+        cycle (24*60*60) and divide by 2π.
+
+    Reference
+    ---------
+    Kalogirou, S. A. (2014). Chapter 2 — Environmental Characteristics. In S. A. Kalogirou (Ed.), Solar Energy Engineering (Second Edition) (pp. 51–123). Academic Press. https://doi.org/10.1016/B978-0-12-397270-5.00002-9
+    """
+    lat = convert_units_to(lat, "rad")
+    h_s = np.arccos(
+        -np.tan(lat) * np.tan(declination)
+    )  # hour angle of sunset (eq. 2.15)
+    # The following equation is not explictely stated in the reference but it can easily be derived.
+    if stat == "integral":
+        return 2 * (
+            h_s * np.sin(declination) * np.sin(lat)
+            + np.cos(declination) * np.cos(lat) * np.sin(h_s)
+        )
+    if stat == "average":
+        return (
+            np.sin(declination) * np.sin(lat)
+            + np.cos(declination) * np.cos(lat) * np.sin(h_s) / h_s
+        )
+
+
+def extraterrestrial_solar_radiation(
+    times: xr.DataArray,
+    lat: xr.DataArray,
+    solar_constant: str = "1361 W m-2",
+    method="spencer",
+) -> xr.DataArray:
+    """Extraterrestrial solar radiation
+
+    This is the daily energy received on a surface parallel to the ground at the mean
+    distance of the earth to the sun. It neglects the effect of the atmosphere. Computation
+    is based on [Kalogirou14]_ and the default solar constant is taken from [Matthes17]_.
+
+    Parameters
+    ----------
+    times: xr.DataArray
+            Daily datetime data. This function makes no sense with data of other frequency.
```

Comment:
```suggestion
        Daily datetime data. This function makes no sense with data of other frequency.
```
Worth checking the frequency in this case, and raise an Error if not daily ?

Reply:
We never do that for indices, so I don't think we should do it for helper function. Yes user may receive wrong data, but if they used a helper function directly instead of an indicator, I assume they made the choice to "bypass" these checks.